# DP: CTRL_CHAR macro: Don't assume character is signed by default

--- ./chardefs.h~	Thu Sep 25 16:59:00 1997
+++ ./chardefs.h	Thu Dec 23 11:18:17 1999
@@ -51,7 +51,7 @@
 #define meta_character_bit 0x080	    /* x0000000, must be on. */
 #define largest_char 255		    /* Largest character value. */
 
-#define CTRL_CHAR(c) ((c) < control_character_threshold && (c) >= 0)
+#define CTRL_CHAR(c) ((c) < control_character_threshold && ((signed char)(c)) >= 0)
 #define META_CHAR(c) ((c) > meta_character_threshold && (c) <= largest_char)
 
 #define CTRL(c) ((c) & control_character_mask)

